### ISSUE TITLE
dotnet clean at the sln level should clean all

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/cli#7240]
## Steps to reproduce
`dotnet clean`

## Expected  behavior
All bin and obj folders for all referenced projects should be cleaned.

## Actual behavior
Nothing seems to happen.

## Environment data
`dotnet --info` output:

```
Product Information:
 Version:            2.0.0-preview3-006770
 Commit SHA-1 hash:  78e7163819

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0-preview3-006770\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0-preview3-25514-02
  Build    : 1b047a04057cec973cfb890fd21f7a639cfcd118
```

**Comment by mairaw** [comment]
I also did `dotnet clean -c Release -r win-x64` and the contents under \bin\Release\netcoreapp2.0\win-x64 were cleaned, but the publish folder under it still had contents. Is that by design?

**Comment by RehanSaeed** [comment]
`dotnet clean` in the 2.0 release only seems to empty the `bin\Debug\netstandard1.6` folder. Ideally it should delete the `bin` and `obj` folders.

**Comment by sharwell** [comment]
> Ideally it should delete the bin and obj folders.

This would be a big problem in a sense similar to NuGet/Home#.

**Comment by livarcocc** [comment]
@nguerrera @dsplaisted it seems clean does not delete the restore files (assets.json, generated props/targets, etc). Do you guys remember if this was intentional?

Clean also does not clean up the publish folder, which in my opinion is a bug.

As for the obj/bin folder, they will not be deleted by clean. They are not deleted in the case of full framework old style csproj either.

**Comment by nguerrera** [comment]
Cleaning restore assets would be up to nuget and is tracked by https://github.com/NuGet/Home/issues/4476 There is some debate there about whether this should be done or not, but that is where it should be decided. We don't generate these files so we should not clean them. 

As for cleaning the publish folder, that is tricky as well. Since build and publish are separate operations, they don't have the same list of files that were written. In some sense. you'd need a `CleanPublish` operation and a fully separate list of files for incremental clean. If we just tack on the publish files on to the list, then if you go build, publish, build, the second build will think that the publish files should be deleted.

**Comment by Petermarcu** [comment]
Yes, builds should give you runnable apps and not half baked apps. Cleans should clean builds and publishes. :)

**Comment by Petermarcu** [comment]
Of course, we have to find a way to do that without giving in on performance.

**Comment by NeelBhatt** [comment]
Hello, Is there any update on this? Will dotnet clean only clear the netstandard2.1 folder?

**Comment by nguerrera** [comment]
> Hello, Is there any update on this? 

No change here.

> Will dotnet clean only clear the netstandard2.1 folder?

I'm not sure I understand the question. 

`dotnet clean` runs the Clean msbuild target, which by design never wholesale clears any files, it deletes the FileWrite items that were recorded during the build. Above discussion demonstrates why this design is problematic for the desire to clean restore or publish assets.

If possible for your workflow, consider using `git clean` (or the equivalent for other source control systems) to get back to a pristine working copy when that is absolutely necessary.

**Comment by NeelBhatt** [comment]
Thanks for the information @nguerrera. Sorry for the confusion but I meant deleting the things which are recorded during the build instead of clearing the files.

**Comment by NinoFloris** [comment]
So why is this about which files are cleaned now? The original issue was about the fact dotnet clean doesn't work on an sln like build can, it's a hindrance and it pushes people to just `rm -r **/obj **/bin`

**Comment by RicardoCostaPS** [comment]
Who ever decided that the clean command wouldn't remove bin folders, deserves a promotion at Microsoft, great job, was it that hard to copy functionality from msbuild ? sad
